docs/src/tutorials/tutorial_4.md

```diff
@@ -0,0 +1,310 @@
+
+Let's continue with a bit more before we draw part of the mascot.
+
+The following actions will just be added to the `javis` function.
+
+```julia
+Action(16:150, (args...)->circle(O, 100, :stroke); subactions=[
+    SubAction(1:15, appear(:fade)),
+]),
+```
+
+This is very similar to the previous action. Here we can see that `SubAction` uses relative frame numbers such that the head appears in the frames `16:30` and then is at full opacity afterwards.
+
+Just a small refresher: We need the anonymous function `(args...)->circle(O, 100, :stroke)` as each function gets called with the three arguments `video, action, frame`.
+
+Okay let's add some hair shall we?
+
+I want to have some randomness in his hair so let's define:
+
+```julia
+hair_angle = rand(-0.9:0.1:0.9, 20)
+```
+
+at the beginning of the `face` function.
+
+and have a hair function:
+
+```julia
+function hair_blob(angle)
+    sethue("brown")
+    rotate(angle)
+    circle(Point(0, -100), 20, :fill)
+end
+```
+
+It draws one brown hair blob given the angle. We basically rotate the whole canvas and then draw the circle always at the same local position. 
+
+Now how do we draw the hair now without creating an action for each blob?
+
+We can use splatting for that :wink:
+
+```julia
+[
+    Action(26:150, (args...)->hair_blob(hair_angle[i]); subactions=[
+        SubAction(1:25, appear(:fade)),
+    ]) for i=1:20
+]...,
+```
+
+I think you get the idea of how to use `appear` now. Let's add some eyes and a nose quickly before we draw our first gif.
+
+```julia
+Action(30:150, (args...)->eyes(eye_centers, 10, "darkblue"); subactions=[
+    SubAction(1:15, appear(:fade)),
+]),
+Action(45:150, (args...)->poly(nose, :fill); subactions=[
+    SubAction(1:15, appear(:fade)),
+]),
+```
+
+with:
+
+```julia
+eye_centers = [Point(-40,-30), Point(40,-30)]
+nose = [O, Point(-10,20), Point(10, 20), O]
+```
+
+and
+
+```julia
+function eyes(centers, radius, color)
+    sethue(color)
+    circle.(centers, radius, :fill)
+    setcolor("white")
+    circle.(centers, radius/5, :fill)
+end
+```
+
+![Up to the nose](./assets/jarvis_nose.gif)
+
+## Using Transformations
+
+Let's give him some moving lips so he can communicate with us:
+
+```julia
+upper_lip = [Point(-40, 45), Point(40, 45)]
+lower_lip = [Point(-40, 55), Point(40, 55)]
+```
+
+These are just the outer points of the lips:
+
+```julia
+function lip(p1, p2)
+    setline(2)
+    move(p1)
+    c1 = p1 + Point(10, 10)
+    c2 = p2 + Point(-10, 10)
+    curve(c1, c2, p2)
+    do_action(:stroke)
+end
+```
+
+This function uses some more functions of the awesome Luxor package.
+
+The lips should be a little thicker than the other lines that we have drawn so far so let's set `setline(2)`. (default is 1).
+First we move to the starting point of the lip and create two control points a bit below and to the vertical center.
+
+The `curve` function is used to draw a bezier curve. 
```

Comment:
I'd suggest maybe a small note on a bezier curve that links to the wikipedia article on bezier curves if anyone has questions.

Reply:
Good point.